library/src/scene_impl.cxx

```
@@ -329,6 +331,109 @@ scene& scene_impl::clear()
   return *this;
 }
 
+//----------------------------------------------------------------------------
+int scene_impl::addLight(const light_state_t& lightState)
+{
+  vtkNew<vtkLight> newLight;
+  newLight->SetLightType(static_cast<int>(lightState.type));
+  newLight->SetPosition(lightState.position[0], lightState.position[1], lightState.position[2]);
+  newLight->SetColor(lightState.color[0], lightState.color[1], lightState.color[2]);
+  newLight->SetPositional(lightState.positionalLight);
+  newLight->SetFocalPoint(lightState.position[0] + lightState.direction[0],
+    lightState.position[1] + lightState.direction[1],
+    lightState.position[2] + lightState.direction[2]);
+  newLight->SetIntensity(lightState.intensity);
+  newLight->SetSwitch(lightState.switchState);
+  this->Internals->Window.GetRenderer()->AddLight(newLight);
+  return getLightCount() - 1;
+}
+
+//----------------------------------------------------------------------------
+int scene_impl::getLightCount() const
+{
+  vtkLightCollection* lc = this->Internals->Window.GetRenderer()->GetLights();
+  const vtkLight* light;
+  vtkCollectionSimpleIterator it;
+  int lightCount = 0;
+  for (lc->InitTraversal(it); (light = lc->GetNextLight(it));)
+  {
+    lightCount++;
+  }
+  return lightCount;
+}
+
+//----------------------------------------------------------------------------
+light_state_t scene_impl::getLight(int index)
+{
+  vtkLightCollection* lc = this->Internals->Window.GetRenderer()->GetLights();
+  vtkLight* light = vtkLight::SafeDownCast(lc->GetItemAsObject(index));
+  if (!light)
+  {
+    log::warn("No light at index ", index, " to get");
+    return light_state_t{};
```

I think this should throw an exception